climada/hazard/relative_cropyield.py

```diff
@@ -596,89 +635,113 @@ def init_hazard_set(filenames, input_dir=INPUT_DIR, bbox=BBOX, isimip_run='ISIMI
+                                'endyear': int(endyear), 'combi_crop': combi_crop,
+                                'combi_crop_irr': f'{combi_crop}-{irr}'}
             his_file_list.append(file)
         elif isimip_run == 'test_file':
             ag_model, cl_model, _, _, soc, co2, crop_prop, *_ = file.split('_')
             _, crop, irr = crop_prop.split('-')
+            combi_crop = crop
             his_file_list.append(file)
             startyear, endyear = yearrange_his
             file_props[file] = {'ag_model': ag_model, 'cl_model': cl_model, 'soc':soc,
                                 'scenario': 'test_file', 'co2':co2, 'crop': crop, 'irr': irr,
                                 'startyear': startyear, 'endyear': endyear,
-                                'crop_irr': crop+'-'+irr}
+                                'crop_irr': f'{crop}-{irr}', 'combi_crop': combi_crop,
+                                'combi_crop_irr': f'{combi_crop}-{irr}'}
+        else:
+            raise ValueError(f'Invalid value for isimip_run: {isimip_run}')
 
-        crop_irr = crop + '-' + irr
-        if crop_irr not in crop_list:
-            crop_list.append(crop_irr)
+        if f'{crop}-{irr}' not in crop_list:
+            crop_list.append(f'{crop}-{irr}')
+        if f'{combi_crop}-{irr}' not in combi_crop_list:
+            combi_crop_list.append(f'{combi_crop}-{irr}')
 
     # generate hazard using the first file to determine the size of the historic mean
     # file structure: ag_model _ cl_model _ scenario _ soc _ co2 _
     #   yield-crop-irr _ fn_str_var _ startyear _ endyear . nc
     #e.g. gepic_gfdl-esm2m_ewembi_historical_2005soc_co2_yield-whe-noirr_
     #   global_annual_1861_2005.nc
     haz_dummy = RelativeCropyield()
-    haz_dummy.set_from_single_run(input_dir=input_dir, filename=his_file_list[0], bbox=bbox,
-                                  scenario=(file_props[his_file_list[0]])['scenario'],
-                                  yearrange=np.array([(file_props[his_file_list[0]])['startyear'],
-                                                      (file_props[his_file_list[0]])['endyear']]))
+    haz_dummy.set_from_isimip_netcdf(input_dir=input_dir, filename=his_file_list[0], bbox=bbox,
+                                  scenario=file_props[his_file_list[0]]['scenario'],
+                                  yearrange=(file_props[his_file_list[0]]['startyear'],
+                                              file_props[his_file_list[0]]['endyear']))
 
     # initiate the historic mean for each combination of crop and irrigation type
     # the idx keeps track of the row in which the hist_mean values are written per crop-irr to
     # ensure that all files are assigned to the corresponding crop-irr combination
     hist_mean_per_crop = dict()
-    for crop_irr in crop_list:
-        crop, irr = crop_irr.split('-')
+    for i, combi_crop_irr in enumerate(combi_crop_list):
+        crop, irr = crop_list[i].split('-')
```

Comment:
Why does `crop_list` contain something else than crops?

```suggestion
    for crops, combi_crop_irr in zip(crop_list, combi_crop_list):
        crop, irr = crops.split('-')
```

Reply:
The variable names are confusing here, because the term "combination" is used for two different kinds of combinations...
combined crops are for instance ri1 and ri2 (two seasons of rice) combined to "ric", while a crop-irr combination is for instance "ric-firr"
```
            crop_list (list): list of all crop-irr combinations
            combi_crop_list (list): list of all crop-irr combinations for combined crops
```
how about renaming the variables as following?
```
            crop_irr_list (list): list of all crop-irr combinations
            combi_crop_irr_list (list): list of all crop-irr combinations for combined crops
```

Reply:
good idea.